plugins/cua/vendor/cua-driver/source/Tests/integration/test_hermes_form_fill_qwen.py

```
@@ -0,0 +1,223 @@
+class _QwenFormBase(unittest.TestCase):
+    """Shared setup/teardown for Qwen-backed hermes form-fill tests."""
+
+    _focus_proc: subprocess.Popen
+    _focus_pid: int
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        # Verify the local server is reachable before spending time on setup.
+        import urllib.request, urllib.error
+        try:
+            urllib.request.urlopen(_ENDPOINT.rstrip("/") + "/models", timeout=3)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)